File: app/src/test/java/com/zl/weilu/androidut/junit/AssertThatTest.java
package com.zl.weilu.androidut.junit;

import org.junit.Rule;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.*;
import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

/**
 * Created by weilu on 2017/10/15.
 */

public class AssertThatTest {

    @Rule
    public MyRule rule = new MyRule();

    @Test
    public void testMobilePhone() throws Exception {
        assertThat("13588888888", new IsMobilePhoneMatcher());
    }

    @Test
    public void testAssertThat1() throws Exception {
        assertThat(6, is(6));
    }

    @Test
    public void testAssertThat2() throws Exception {
        assertThat(null, nullValue());
    }

    @Test
    public void testAssertThat3() throws Exception {
        assertThat("Hello UT", both(startsWith("Hello")).and(endsWith("UT")));
    }
}

File: app/src/test/java/com/zl/weilu/androidut/junit/AssertThatTest.kt
package com.zl.weilu.androidut.junit

import org.hamcrest.CoreMatchers.*
import org.hamcrest.core.Is.`is`
import org.junit.Assert.assertThat
import org.junit.Rule
import org.junit.Test

/**
 * Created by weilu on 2017/10/15.
 */

class AssertThatTest {

    @get:Rule
    var rule = MyRule()

    @Test
    @Throws(Exception::class)
    fun testMobilePhone() {
        assertThat("13588888888", IsMobilePhoneMatcher())
    }

    @Test
    @Throws(Exception::class)
    fun testAssertThat1() {
        assertThat(6, `is`(6))
    }

    @Test
    @Throws(Exception::class)
    fun testAssertThat2() {
        assertThat<Any>(null, nullValue())
    }

    @Test
    @Throws(Exception::class)
    fun testAssertThat3() {
        assertThat("Hello UT", both(startsWith("Hello")).and(endsWith("UT")))
    }
}
